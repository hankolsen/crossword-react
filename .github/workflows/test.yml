name: melodikryzzet-tests

on: 
  push:
    branches:
      - 'master'
  pull_request:


jobs:
  test:
    name: 'Node ${{ matrix.node }}'
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node: [12, 14]
    env:
      SKIP_PREFLIGHT_CHECK: true
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with: 
        node-version: ${{ matrix.node }}
    - name: Install dependencies
      uses: bahmutov/npm-install@v1
      with: 
        install-command: yarn
    - name: Run tests
      run: yarn test:ci