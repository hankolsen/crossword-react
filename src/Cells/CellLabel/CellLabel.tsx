import React from 'react';
import { CELL_HEIGHT, CELL_WIDTH } from 'config';

type Props = {
  row: number;
  column: number;
  number: number;
};

const CellLabel = ({ row, column, number }: Props) => (
  <text
    x={2 + column * (CELL_WIDTH + 1)}
    y={10 + row * (CELL_HEIGHT + 1)}
    className="crossword__cell-number"
  >
    {number}
  </text>
);

export default CellLabel;
