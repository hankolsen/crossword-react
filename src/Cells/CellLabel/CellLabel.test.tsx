import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from 'test/test-utils';
import CellLabel from './CellLabel';

describe('CellLabel test', () => {
  it('should render correctly', async () => {
    const { container } = render(
      <svg>
        <CellLabel row={1} column={1} number={8} />
      </svg>,
    );
    expect(await screen.findByText('8')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
