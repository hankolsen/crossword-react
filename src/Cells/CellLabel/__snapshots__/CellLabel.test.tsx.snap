// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellLabel test should render correctly 1`] = `
<div>
  <div>
    <svg>
      <text
        class="crossword__cell-number"
        x="34"
        y="42"
      >
        8
      </text>
    </svg>
  </div>
</div>
`;
